src/pages/home/report/ReportActionCompose.js

```diff
@@ -66,18 +72,24 @@ class ReportActionCompose extends React.Component {
         this.setIsFocused = this.setIsFocused.bind(this);
         this.focus = this.focus.bind(this);
         this.comment = props.comment;
+        this.autoFocusEnabled = getPlatform() === 'desktop' || (getPlatform() === 'web' && !canUseTouchScreen());
 
         this.state = {
-            isFocused: true,
+            isFocused: this.autoFocusEnabled,
             textInputShouldClear: false,
             isCommentEmpty: props.comment.length === 0,
             isMenuVisible: false,
         };
     }
 
     componentDidUpdate(prevProps) {
-        // When any modal goes from visible to hidden, bring focus to the compose field
-        if (prevProps.modal.isVisible && !this.props.modal.isVisible) {
+        // We want to manually set the focus only when.
+        //  1) Auto focus is enabled(we are on desktop/Desktop Web).
+        //  1) Screen is focsued.
+        //  2) When any modal goes from visible to hidden &
```

Comment:
This comment needs to be cleaned up a bit and formatted. The numbers are 1, 1, and 2, focused is spelled wrong, and there is a trailing `&`

It also doesn't really help explain why we are doing all of this. So I think we can just change it to...

>We only want to focus or refocus the input when a modal has been closed and the underlying screen is focused. We avoid doing this on native platforms since the software keyboard popping open creates a jarring and broken UX.

I think this way, someone will think twice before modifying the logic here.

Reply:
Haha, Sure. Feels like I need to work on better explanations more than code. :rofl: